pandas/tests/arrays/boolean/test_arithmetic.py

```diff
@@ -13,30 +15,87 @@ def data():
     )
 
 
-class TestArithmeticOps(BaseOpsUtil):
-    def test_error(self, data, all_arithmetic_operators):
-        # invalid ops
+@pytest.fixture
+def a():
+    return pd.array([True] * 3 + [False] * 3 + [None] * 3, dtype="boolean")
 
-        op = all_arithmetic_operators
-        s = pd.Series(data)
-        ops = getattr(s, op)
-        opa = getattr(data, op)
 
-        # invalid scalars
-        with pytest.raises(TypeError):
-            ops("foo")
-        with pytest.raises(TypeError):
-            ops(pd.Timestamp("20180101"))
+@pytest.fixture
+def b():
+    return pd.array([True, False, None] * 3, dtype="boolean")
+
 
-        # invalid array-likes
-        if op not in ("__mul__", "__rmul__"):
-            # TODO(extension) numpy's mul with object array sees booleans as numbers
-            with pytest.raises(TypeError):
-                ops(pd.Series("foo", index=s.index))
+# Basic test for the arithmetic array ops
+# -----------------------------------------------------------------------------
 
-        # 2d
-        result = opa(pd.DataFrame({"A": s}))
-        assert result is NotImplemented
 
-        with pytest.raises(NotImplementedError):
-            opa(np.arange(len(s)).reshape(-1, len(s)))
+@pytest.mark.parametrize(
+    "opname, exp",
+    [
+        ("add", [True, True, None, True, False, None, None, None, None]),
+        ("mul", [True, False, None, False, False, None, None, None, None]),
+    ],
+    ids=["add", "mul"],
+)
+def test_add_mul(a, b, opname, exp):
+    op = getattr(operator, opname)
+    result = op(a, b)
+    expected = pd.array(exp, dtype="boolean")
+    tm.assert_extension_array_equal(result, expected)
+
+
+def test_sub(a, b):
+    with pytest.raises(TypeError):
+        # numpy points to ^ operator or logical_xor function instead
+        a - b
+
+
+def test_div(a, b):
+    # for now division gives a float numpy array
+    result = a / b
+    expected = np.array(
+        [1.0, np.inf, np.nan, 0.0, np.nan, np.nan, np.nan, np.nan, np.nan],
+        dtype="float64",
+    )
+    tm.assert_numpy_array_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "opname",
+    [
+        "floordiv",
+        "mod",
+        pytest.param(
+            "pow", marks=pytest.mark.xfail(reason="TODO follow int8 behaviour?")
+        ),
+    ],
+)
+def test_op_int8(a, b, opname):
+    op = getattr(operator, opname)
+    result = op(a, b)
+    expected = op(a.astype("Int8"), b.astype("Int8"))
+    tm.assert_extension_array_equal(result, expected)
+
+
+# Test generic charachteristics / errors
```

Comment:
sp?

Reply:
Does "sp" point to the typo in this title? Or did you mean something else?